buildlib/tools/coverity.sh

```diff
@@ -44,31 +44,44 @@ modules_for_coverity_unload() {
 	return $res
 }
 
-run_coverity() {
-
-	az_init_modules
-	modules_for_coverity
-
+configure_coverity() {
 	ucx_build_type=$1
 
 	xpmem_root=$(module show $XPMEM_MODULE 2>&1 | awk '/CPATH/ {print $3}' | sed -e 's,/include,,')
 	with_xpmem="--with-xpmem=$xpmem_root"
 
-	${WORKSPACE}/contrib/configure-$ucx_build_type --prefix=$ucx_inst --with-cuda --with-gdrcopy --with-java $with_xpmem
+	if [ "${ENABLE_JAVA:-yes}" = yes ]; then
+		java_opt="--with-java"
+	else
+		java_opt="--without-java"
+	fi
```

Comment:
I think that if we're adapting this script to usage without CI, make it more generic:
- splitting build/analyze is good step
- instead of using ENABLE_JAVA as env var, pass build options and control options using command line args. and document the usage with --help
- avoid checking INCREMENTAL env var; just let users set env vars outside the script

Reply:
i think for coverity one coverity-build needs one analyze so we don't generally need to analyze repeatedly on a given build. what about doing like below:

```
$ coverity.sh release --no-clean --no-java
$ coverity.sh release
$ coverity.sh devel

$ coverity.sh
$ coverity.sh help
coverity.sh <release|devel> [--no-clean] [--no-java]
  --no-clean     reuse the existing build
  --no-java      do not enable java in the build
```

Reply:
Maybe "--clean" (opt-in instead of opt-out)
instead of --no-java we can pass custom config args (--with-java or --without-java if needed; if neither is passed, configure will use guess as default mode)

Reply:
makes sense, something like:
```
$ coverity.sh help
coverity.sh <release|devel> [--clean] [<configure opts>]
  --clean     reuse the existing build
```

Reply:
yeah, every unrecognized option (e.g not --clean) is passed to configure